src/examples/euler/EulerSingleSidedHook.sol

```diff
@@ -0,0 +1,126 @@
+
+import {IERC20} from "forge-std/interfaces/IERC20.sol";
+import {IPoolManager} from "v4-core/src/interfaces/IPoolManager.sol";
+
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+import {PoolKey} from "v4-core/src/types/PoolKey.sol";
+import {Currency} from "v4-core/src/types/Currency.sol";
+import {TickMath} from "v4-core/src/libraries/TickMath.sol";
+
+import {JIT} from "../../JIT.sol";
+import {JITHook} from "../../JITHook.sol";
+
+import {EulerUtils} from "./EulerUtils.sol";
+
+/// @title Euler Account JIT Hook
+/// @notice Pulls funds from a specified Euler vaults through a specific swap account, create a JIT position, and transfers funds back to the swap account
+/// @notice When user is doing exaactIn or exactOut swap, this will JIT single-sided liquidity only
+/// @dev JIT position is 8% width
+contract EulerSingleSidedHook is JITHook, EulerUtils {
+    constructor(address evcAddr, IPoolManager poolManagerAddr, address vaultA, address vaultB, address swapAccountAddr)
+        JITHook(poolManagerAddr)
+        EulerUtils(evcAddr, vaultA, vaultB, swapAccountAddr)
+    {}
+
+    /// @inheritdoc JITHook
+    function _pull(PoolKey calldata key, IPoolManager.SwapParams calldata swapParams)
+        internal
+        override
+        returns (address excessRecipient, uint128 amount0, uint128 amount1)
+    {
+        excessRecipient = swapAccount;
+
+        (, Currency outputCurrency, uint256 amount) = _getInputOutputAndAmount(key, swapParams);
+
+        // transferFrom: depositor's currency0 or currency1 to the PoolManager based on currency-out from the pool
+        if (outputCurrency == key.currency0) {
+            amount0 = _sendToPoolManager(vault0, key.currency0, uint128(amount));
+        } else {
+            amount1 = _sendToPoolManager(vault1, key.currency1, uint128(amount));
+        }
+    }
+
+    /// @inheritdoc JITHook
+    function _recipient() internal view override returns (address) {
+        return swapAccount;
+    }
+
+    /// @dev computes the tick range of the JIT position
+    ///      If providing amount0 of tokens0 only, return ticks current to +8%
+    ///      If providing amount1 of token1 only, return ticks -8% to current
+    /// @inheritdoc JIT
+    function _getTickRange(PoolKey memory poolKey, uint128 amount0, uint128 /*amount1*/, uint160 sqrtPriceX96)
+        internal
+        pure
+        override
+        returns (int24 tickLower, int24 tickUpper)
+    {
+        if (amount0 > 0) {
+            // Only token0: provide liquidity in current to +8% range
+            uint160 _sqrtPriceUpper = uint160(
+                FixedPointMathLib.mulDivDown(
+                    uint256(sqrtPriceX96), FixedPointMathLib.sqrt(1.08e18), FixedPointMathLib.sqrt(1e18)
+                )
+            );
+
+            int24 _tickLowerUnaligned = TickMath.getTickAtSqrtPrice(sqrtPriceX96);
+            int24 _tickUpperUnaligned = TickMath.getTickAtSqrtPrice(_sqrtPriceUpper);
```

Comment:
nice this math checks out quite nicely, just be aware that that `tickLower = getTickAtSqrtPrice(sqrtPrice)` may produce a range thats dual-sided:

i.e. this scenario is dual sided liquidity

```
    tickLower                            tickUpper
<---|---------x-----------|--------------|----->
              |_sqrtPrice
```

Reply:
its probably passing in your tests because `sqrtPrice == tickBoundary` so the JIT position is indeed single-sided. but if the sqrtPrice were to move off the tick boundary, the current math will create dual sided liquidity!